src/bp/ui-shared/src/Toaster/index.tsx

```diff
@@ -3,36 +3,56 @@ import _ from 'lodash'
 
 import { lang } from '../translations'
 
-export enum Timeout {
-  SHORT = 1000,
-  MEDIUM = 3000,
-  LONG = 5000
-}
-
 export interface ToastOptions {
-  delayed: boolean
-  timeout: Timeout
+  // Delaying to avoid the react lifecycle issue (executing before it is rendered)
+  delayed?: boolean
+  timeout?: 'short' | 'medium' | 'long'
   onDismiss?: (didTimeoutExpire: boolean) => void
 }
 
-export const toastFailure = (
+const success = (
   message: string,
   details?: string,
   options: ToastOptions = {
     delayed: false,
-    timeout: Timeout.MEDIUM
+    timeout: 'short'
+  }
+) => showToast(lang(message, { details }), Intent.SUCCESS, options)
+
+const failure = (
+  message: string,
+  details?: string,
+  options: ToastOptions = {
+    delayed: true,
+    timeout: 'medium'
   }
 ) => {
-  toast(lang(message, { details }), Intent.DANGER, options.timeout, options.onDismiss, options)
+  showToast(lang(message, { details }), Intent.DANGER, options)
 }
 
-const toast = (message, intent, timeout, onDismiss, options?) => {
+const info = (
+  message: string,
```

Comment:
Let's say the translation has some keys, what's the strategy here ?  The developer translates before passing to toast ?

For instance a toast info could display something like:
"Hello, you have {n} new messages", how do I pass the n ?

Reply:
Yup, that's what I had in mind. Just to keep it simple for 90% of the use cases, when you have a special case, just translate it before passing it to the message. 

It's useful for the toast failure, when you want to display a custom message, and include the error message

Reply:
Great !